media-tv/gentoo-vdr-scripts/files/systemd-fix-environment.patch

```diff
@@ -0,0 +1,18 @@
+diff -Naur gentoo-vdr-scripts-2.8.orig/etc/sudoers.d/vdr gentoo-vdr-scripts-2.8/etc/sudoers.d/vdr
+--- gentoo-vdr-scripts-2.8.orig/etc/sudoers.d/vdr	2019-12-15 01:04:54.000000000 +0100
++++ gentoo-vdr-scripts-2.8/etc/sudoers.d/vdr	2021-05-29 16:26:34.623489561 +0200
+@@ -1 +1,2 @@
+ vdr ALL=NOPASSWD:/usr/share/vdr/bin/vdrshutdown-really.sh
++vdr ALL=NOPASSWD:/bin/systemctl daemon-reload
+diff -Naur gentoo-vdr-scripts-2.8.orig/usr/share/vdr/systemd/vdr-systemd_helper.sh gentoo-vdr-scripts-2.8/usr/share/vdr/systemd/vdr-systemd_helper.sh
+--- gentoo-vdr-scripts-2.8.orig/usr/share/vdr/systemd/vdr-systemd_helper.sh	2019-12-15 01:04:54.000000000 +0100
++++ gentoo-vdr-scripts-2.8/usr/share/vdr/systemd/vdr-systemd_helper.sh	2021-05-29 17:19:16.810413833 +0200
+@@ -105,7 +105,7 @@
+ 	# as the user under which vdr will run is controlled by systemd
+ 	sed -e "s:'-u' 'vdr' ::" -i ${SYSTEMD_ENV_FILE}
+ 
+-	sync
++	sudo systemctl daemon-reload
```

Comment:
This looks very weird.

Reply:
During service start, the `ExecStartPre=` collects parameters and modifies the environment file, which used by the service started in the next step. Systemd notices this file modification and prints out an error message - unless systemd is explicitely reloaded. 
Still weird?

Reply:
I don't use systemd, but shouldn't that be handled in the service file somehow? What if I don't have sudo configured - this would give sudo rights to the `vdr` user right? It doesn't have a password, that sounds a bit scary.

Reply:
> I don't use systemd, but shouldn't that be handled in the service file somehow? 

Yes thats exactly whats happening there. The `ExecStartPre` prepared the environment for the service, and collects config snippets from plugins to config files.

> What if I don't have sudo configured - this would give sudo rights to the `vdr` user right?

Yes... and no. The package installs a sudoers file with this content:
```
vdr ALL=NOPASSWD:/usr/share/vdr/bin/vdrshutdown-really.sh
vdr ALL=NOPASSWD:/bin/systemctl daemon-reload
```
so thats exactly what user "vdr" is allow to execute with root rights.


> It doesn't have a password, that sounds a bit scary.

Maybe, but I see no other way to get this done.